### PR TITLE
Added new vocabulary elements

### PR DESCRIPTION
**Pull request by alien-mcl**
Added new vocabulary elements as in #41 and #16

---
This change is [Reviewable](https://reviewable.io/reviews/hydracg/specifications/157)

### CODE REVIEW

File: spec/latest/core/core.jsonld [hunk 1]
```diff
@@ -299,6 +302,27 @@
...
+    {
+      "@id": "hydra:collection",
+      "@type": "hydra:Link",
+      "label": "collection",
+      "comment": "Allows to link multiple collections to an owning resource.",
+      "domain": "hydra:Collection",
+      "vs:term_status": "testing"
+    },
+    {
+      "@id": "hydra:manages",
```
Comment (tpluscode) [inline spec/latest/core/core.jsonld]: Do we have enough consensus about the manages block?

Comment (alien-mcl) [reply spec/latest/core/core.jsonld]: Well, #41 is marked as resolved with `hydra:manages` as one of the solutions. While I think it indeed solves some problems, I've got personal issues with `hydra:collection` as I don't see any reason for making a special distinction between ordinary link and collection link (as this is how I understand `hydra:collection`)

**Comment by lanthaler** [comment]
Review status: 0 of 1 files reviewed at latest revision, 5 unresolved discussions.

---

*[spec/latest/core/core.jsonld, line 314 at r1](https://reviewable.io:443/reviews/hydracg/specifications/157#-L7TFzllEv7mowjAT0a_-r1-314:-L7TH0RKp9wG-3jCmU4F:b-txmw0k) ([raw file](https://github.com/hydracg/specifications/blob/fa9f921099c077c5ecebaf84c80e8d0a6c77629f/spec/latest/core/core.jsonld#L314)):*
<details><summary><i>Previously, alien-mcl (Karol Szczepański) wrote…</i></summary><blockquote>

Well, #41 is marked as resolved with `hydra:manages` as one of the solutions. While I think it indeed solves some problems, I've got personal issues with `hydra:collection` as I don't see any reason for making a special distinction between ordinary link and collection link (as this is how I understand `hydra:collection`)
</blockquote></details>

Yep, there was a call for consensus with no objections

---

*[spec/latest/core/core.jsonld, line 37 at r2](https://reviewable.io:443/reviews/hydracg/specifications/157#-L7TGDtVei7XRPeTC5KR:-L7TGDtVei7XRPeTC5KS:b-6lmo23) ([raw file](https://github.com/hydracg/specifications/blob/fec61202e499babbb3163063b674fabb2df67558/spec/latest/core/core.jsonld#L37)):*
> ```JSONLD
>     "Collection": "hydra:Collection",
>     "member": { "@id": "hydra:member", "@type": "@id" },
>     "manages": { "@id": "hydra:manages", "@type": "@id" },
> ```

We can simplify this to simply `"hydra:manages"`

---

*[spec/latest/core/core.jsonld, line 308 at r2](https://reviewable.io:443/reviews/hydracg/specifications/157#-L7TGe1RfkrVqKIqdZpa:-L7TGe1RfkrVqKIqdZpb:byb62no) ([raw file](https://github.com/hydracg/specifications/blob/fec61202e499babbb3163063b674fabb2df67558/spec/latest/core/core.jsonld#L308)):*
> Allows to link multiple collections to an owning resource.

Please change this to something like "Collections somehow related to this resource."

---

*[spec/latest/core/core.jsonld, line 322 at r2](https://reviewable.io:443/reviews/hydracg/specifications/157#-L7TH8Ol6lKqeotzdVOL:-L7TH8Ol6lKqeotzdVOM:be851uv) ([raw file](https://github.com/hydracg/specifications/blob/fec61202e499babbb3163063b674fabb2df67558/spec/latest/core/core.jsonld#L322)):*
> "Provides details on on the resource being a subject in relation described by the collection."

As we might re-use this elsewhere at some point I'd suggest to change it to simply "The subject."

---

*[spec/latest/core/core.jsonld, line 324 at r2](https://reviewable.io:443/reviews/hydracg/specifications/157#-L7THQA_0Gtj93jo2abC:-L7THQA_0Gtj93jo2abD:bgigmx5) ([raw file](https://github.com/hydracg/specifications/blob/fec61202e499babbb3163063b674fabb2df67558/spec/latest/core/core.jsonld#L324)):*
> ```JSONLD
>       "comment": "Provides details on on the resource being a subject in relation described by the collection.",
>       "vs:term_status": "testing"
>     },
> ```

Looks like you forgot to add `hydra:object`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hydracg/specifications/157)*

**Comment by alien-mcl** [comment]
Review status: 0 of 1 files reviewed at latest revision, 5 unresolved discussions.

---

*[spec/latest/core/core.jsonld, line 37 at r2](https://reviewable.io:443/reviews/hydracg/specifications/157#-L7TGDtVei7XRPeTC5KR:-L7Vn43qbxGa_L33cSgs:bqxgtuc) ([raw file](https://github.com/hydracg/specifications/blob/fec61202e499babbb3163063b674fabb2df67558/spec/latest/core/core.jsonld#L37)):*
<details><summary><i>Previously, lanthaler (Markus Lanthaler) wrote…</i></summary><blockquote>

We can simplify this to simply `"hydra:manages"`
</blockquote></details>

Are you sure? Indeed when the value of this relation will be a JSON-object, expansion will work. But I've noticed that string literals are not correctly interpreted as resource IRIs. I know, that in most cases there will be an object here, but still, corner case may get broken.

---

*[spec/latest/core/core.jsonld, line 308 at r2](https://reviewable.io:443/reviews/hydracg/specifications/157#-L7TGe1RfkrVqKIqdZpa:-L7VoSJGYgFocIF3uKGA:b-896fix) ([raw file](https://github.com/hydracg/specifications/blob/fec61202e499babbb3163063b674fabb2df67558/spec/latest/core/core.jsonld#L308)):*
<details><summary><i>Previously, lanthaler (Markus Lanthaler) wrote…</i></summary><blockquote>

> Allows to link multiple collections to an owning resource.

Please change this to something like "Collections somehow related to this resource."
</blockquote></details>

Done.

---

*[spec/latest/core/core.jsonld, line 322 at r2](https://reviewable.io:443/reviews/hydracg/specifications/157#-L7TH8Ol6lKqeotzdVOL:-L7VodIG3E3DifO4dKVc:b-896fix) ([raw file](https://github.com/hydracg/specifications/blob/fec61202e499babbb3163063b674fabb2df67558/spec/latest/core/core.jsonld#L322)):*
<details><summary><i>Previously, lanthaler (Markus Lanthaler) wrote…</i></summary><blockquote>

> "Provides details on on the resource being a subject in relation described by the collection."

As we might re-use this elsewhere at some point I'd suggest to change it to simply "The subject."
</blockquote></details>

Done.

---

*[spec/latest/core/core.jsonld, line 324 at r2](https://reviewable.io:443/reviews/hydracg/specifications/157#-L7THQA_0Gtj93jo2abC:-L7VohfSGVlfuLSII_vT:be4gbx1) ([raw file](https://github.com/hydracg/specifications/blob/fec61202e499babbb3163063b674fabb2df67558/spec/latest/core/core.jsonld#L324)):*
<details><summary><i>Previously, lanthaler (Markus Lanthaler) wrote…</i></summary><blockquote>

Looks like you forgot to add `hydra:object`
</blockquote></details>

Hmm. Honestly - I though it was enlisted mistakenly.

I acknowledged `hydra:manages` block is to re-instantiate the relation between the resource owning a given collection (i.e. resource being on the left of the `hydra:collection` predicate) and the resources provided by the collection. Having an explicit `hydra:object` here could prove difficult for interpretation on what actually that owning resource is doing here in case `hydra:object` does not point to that owning resource.

Also having all subject-property-object components, why having them repeated in the Hydra vocab instead of using rdf reification ones?

I've added it, as the #41 states it as a solution, but I'm not convinced we need that.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hydracg/specifications/157)*

**Comment by tpluscode** [comment]
Review status: 0 of 1 files reviewed at latest revision, 4 unresolved discussions.

---

*[spec/latest/core/core.jsonld, line 37 at r2](https://reviewable.io:443/reviews/hydracg/specifications/157#-L7TGDtVei7XRPeTC5KR:-L7pf1LlyTnLpRqcie4a:b-y5d35) ([raw file](https://github.com/hydracg/specifications/blob/fec61202e499babbb3163063b674fabb2df67558/spec/latest/core/core.jsonld#L37)):*
<details><summary><i>Previously, alien-mcl (Karol Szczepański) wrote…</i></summary><blockquote>

Are you sure? Indeed when the value of this relation will be a JSON-object, expansion will work. But I've noticed that string literals are not correctly interpreted as resource IRIs. I know, that in most cases there will be an object here, but still, corner case may get broken.
</blockquote></details>

I agree with Karol. `"manages": "some:url"` should be interpreted correctly

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hydracg/specifications/157)*

**Comment by tpluscode** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hydracg/specifications/157)*

**Comment by lanthaler** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[spec/latest/core/core.jsonld, line 37 at r2](https://beta.reviewable.io/reviews/hydracg/specifications/157#-L7TGDtVei7XRPeTC5KR:-L7tvxNKAtbi3WfbnQPQ:b-adj94r) ([raw file](https://github.com/hydracg/specifications/blob/fec61202e499babbb3163063b674fabb2df67558/spec/latest/core/core.jsonld#L37)):*
<details><summary><i>Previously, tpluscode (Tomasz Pluskiewicz) wrote…</i></summary><blockquote>

I agree with Karol. `"manages": "some:url"` should be interpreted correctly
</blockquote></details>

I'd fine it very surprising to find URLs as the value of this. So I prefer this to fail in a predictable and easy to spot manner.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/hydracg/specifications/157)*

**Comment by tpluscode** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[spec/latest/core/core.jsonld, line 37 at r2](https://beta.reviewable.io/reviews/hydracg/specifications/157#-L7TGDtVei7XRPeTC5KR:-L7uRYBv3tpBd_ULxLRQ:b-44pdgl) ([raw file](https://github.com/hydracg/specifications/blob/fec61202e499babbb3163063b674fabb2df67558/spec/latest/core/core.jsonld#L37)):*
<details><summary><i>Previously, lanthaler (Markus Lanthaler) wrote…</i></summary><blockquote>

I'd fine it very surprising to find URLs as the value of this. So I prefer this to fail in a predictable and easy to spot manner.
</blockquote></details>

Oh, the object will be the "manages block", right? Yes, it probably makes little sense to have type coercion here

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/hydracg/specifications/157)*

**Comment by lanthaler** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[spec/latest/core/core.jsonld, line 37 at r2](https://beta.reviewable.io/reviews/hydracg/specifications/157#-L7TGDtVei7XRPeTC5KR:-L8qlSwwCNNVDATEQNPy:b-w15flx) ([raw file](https://github.com/hydracg/specifications/blob/fec61202e499babbb3163063b674fabb2df67558/spec/latest/core/core.jsonld#L37)):*
<details><summary><i>Previously, tpluscode (Tomasz Pluskiewicz) wrote…</i></summary><blockquote>

Oh, the object will be the "manages block", right? Yes, it probably makes little sense to have type coercion here
</blockquote></details>

Yes, the it will be the "manages block"

---

*[spec/latest/core/core.jsonld, line 39 at r3](https://beta.reviewable.io/reviews/hydracg/specifications/157#-L8qlYYe3wtA7_nxipL0:-L8qlYYe3wtA7_nxipL1:b-yinqhd) ([raw file](https://github.com/hydracg/specifications/blob/697bd7aea8decf68e0ba45fb18942b0e87cbae29/spec/latest/core/core.jsonld#L39)):*
> ```JSONLD
>     "manages": { "@id": "hydra:manages", "@type": "@id" },
>     "subject": { "@id": "hydra:subject", "@type": "@id" },
>     "object": { "@id": "hydra:subject", "@type": "@id" },
> ```

`hydra:subject`

This needs to be `hydra:object`. Could you please type-coerce both to `@vocab` (like `property`)

---

*[spec/latest/core/core.jsonld, line 172 at r3](https://beta.reviewable.io/reviews/hydracg/specifications/157#-L8qm--V-2_2HLY7Akhy:-L8qm--V-2_2HLY7Akhz:bkhqybb) ([raw file](https://github.com/hydracg/specifications/blob/697bd7aea8decf68e0ba45fb18942b0e87cbae29/spec/latest/core/core.jsonld#L172)):*
> ```JSONLD
>       "@type": "rdf:Property",
>       "label": "property",
>       "comment": "A property, either supported by a class or described by a collection",
> ```

This just makes the documentation more confusing. I'd suggest to revert this change.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/hydracg/specifications/157)*

**Comment by lanthaler** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/hydracg/specifications/157)*

**Comment by alien-mcl** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[spec/latest/core/core.jsonld, line 39 at r3](https://beta.reviewable.io/reviews/hydracg/specifications/157#-L8qlYYe3wtA7_nxipL0:-L8sEgIZ6Jp9hzkny_Ag:b-896fix) ([raw file](https://github.com/hydracg/specifications/blob/697bd7aea8decf68e0ba45fb18942b0e87cbae29/spec/latest/core/core.jsonld#L39)):*
<details><summary><i>Previously, lanthaler (Markus Lanthaler) wrote…</i></summary><blockquote>

`hydra:subject`

This needs to be `hydra:object`. Could you please type-coerce both to `@vocab` (like `property`)
</blockquote></details>

Done.

---

*[spec/latest/core/core.jsonld, line 172 at r3](https://beta.reviewable.io/reviews/hydracg/specifications/157#-L8qm--V-2_2HLY7Akhy:-L8sEp923Z2gbCrNhpQq:b-896fix) ([raw file](https://github.com/hydracg/specifications/blob/697bd7aea8decf68e0ba45fb18942b0e87cbae29/spec/latest/core/core.jsonld#L172)):*
<details><summary><i>Previously, lanthaler (Markus Lanthaler) wrote…</i></summary><blockquote>

This just makes the documentation more confusing. I'd suggest to revert this change.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/hydracg/specifications/157)*

**Comment by lanthaler** [comment]
Thanks Karol!

---

Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/hydracg/specifications/157#-:-L90aoRFD-gB-xp6t4Mz:bacoli2)*

**Pull request merged**